### DISCUSSION TITLE
Terminal B becomes laggy when compiling in terminal A

### DISCUSSION THREAD

**Discussion by anzhi0708** [microsoft/terminal#17079] [Q&A] [answered]
When opening two terminals, while compiling resource-intensive tasks in terminal A, the other terminal becomes so laggy it's unusable. However, processes outside of the terminals, such as instant messaging tools or browsers, run very smoothly. Is this due to memory/CPU usage limitations of PowerShell or dotnet? If it really is due to the consumption of hardware resources, other processes should lag in the same way.

**Comment by lhecker** [comment, marked as answer]
That depends on a lot of factors. For instance if you configured PowerShell with posh-git it may become laggy because your disk is busy which slows down git. It may be due to your Anti-Virus which scans PowerShell. There's a ton of possibilities why this may happen. The only way to properly answer this is to run a perf trace with WPR (Windows Performance Recorder) but before that it may be worth checking your general system performance first, in particular whether your disk is busy with IO.
